src/main/java/seedu/address/logic/commands/WorkerPayCommand.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.worker.Worker;
+
+
+
+/**
+ * Prints the pay earned by a worker identified using their displayed index.
+ */
+public class WorkerPayCommand extends Command {
+
+    public static final String COMMAND_WORD = "worker-pay";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Prints the pay earned by a worker identified by the index number used in the displayed worker list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_SHOW_PAY_SUCCESS = "Pay earned by worker (%1$s): $ %2$.2f";
+
+    private final Index targetIndex;
+
+    public WorkerPayCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Worker> lastShownList = model.getFilteredWorkerList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_WORKER_DISPLAYED_INDEX);
+        }
+
+        Worker selectedWorker = lastShownList.get(targetIndex.getZeroBased());
+        float calculatedPay = model.calculateWorkerPay(selectedWorker);
```

Comment:
I think implementing the logic of `calculateWorkerPay()` in this class itself may be more consistent with the rest of our commands. Maybe the rest can chip in on this

Reply:
I think this is ok, for a possible future implementation could allow shifts of differing number of hours - hence calculating worker pay will need to be at a model level. Besides `Worker` has no knowledge of its assignments

Reply:
Okay, I'll put `calculateWorkerPay()` in model for now. Putting `calculateWorkerPay()` in the command class is also fine.